src/iot/commands/attachCertificate.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger } from '../../shared/logger'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { IotThingNode } from '../explorer/iotThingNode'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { PromptResult } from '../../shared/ui/prompter'
+import { IotCertificate } from '../../shared/clients/iotClient'
+import { WizardControl } from '../../shared/wizards/wizard'
+import { Iot } from 'aws-sdk'
+
+/**
+ * Attaches a certificate to the thing represented by the given node.
+ *
+ * Prompts the user to select a certificate
+ * Attaches the certificate.
+ * Refreshes the thing node.
+ */
+export async function attachCertificateCommand(
+    node: IotThingNode,
+    promptFun = promptForCert,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('AttachCertificate called for %O', node)
+
+    const thingName = node.thing.name
+
+    let nextToken: string | undefined = undefined
+    let certificates: IotCertificate[] = []
+    do {
+        try {
+            const certResponse: Iot.ListCertificatesResponse = await node.iot.listCertificates({ marker: nextToken })
+            nextToken = certResponse.nextMarker
+
+            const newCerts =
+                certResponse.certificates
+                    ?.filter(cert => cert.certificateArn && cert.certificateId && cert.status && cert.creationDate)
+                    .map(cert => ({
+                        arn: cert.certificateArn!,
+                        id: cert.certificateId!,
+                        activeStatus: cert.status!,
+                        creationDate: cert.creationDate!,
+                    })) ?? []
+
+            certificates = certificates.concat(newCerts)
+        } catch (e) {
+            getLogger().error(`Failed to retrieve certificates: %O`, e)
+            showViewLogsMessage(localize('AWS.iot.attachCert.error', 'Failed to retrieve certificates'), window)
+            return undefined
```

Comment:
I think this will be addressed when I finish using an AsyncIterable for this as per https://github.com/aws/aws-toolkit-vscode/pull/2197#discussion_r734810945

Reply:
Updated to use the async generator